satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -558,13 +586,24 @@ def __init__(self, filename, filename_info, filetype_info,  # noqa: D417
             chunks={"x": CHUNK_SIZE,
                     "y": CHUNK_SIZE,
                     "x_ir_wv": CHUNK_SIZE,
-                    "y_ir_wv": CHUNK_SIZE}
+                    "y_ir_wv": CHUNK_SIZE},
+            decode_cf=False,
+            decode_times=False,
+            mask_and_scale=False,
         )
+
         self.nc = DatasetWrapper(nc_raw)
+        # decode times and fix duplicate dimensions
+        self.nc.prepare_input()
 
         # Projection longitude is not provided in the file, read it from the
         # filename.
-        self.projection_longitude = float(filename_info["projection_longitude"])
+        if "." in str(filename_info["projection_longitude"]):
+            self.projection_longitude = float(filename_info["projection_longitude"])
+        else:
+            self.projection_longitude = (
+                float(filename_info["projection_longitude"][:2] + "." + filename_info["projection_longitude"][2:])
+            )
```

Comment:
This could be moved to a separate method `get_projection_longitude`.

Reply:
In the second case, I think you could do 

```python
float(filename_info["projection_longitude"]) / 100
```

Reply:
sounds good